File: example/customersdotnet/Interfaces.cs
using System.Threading.Tasks;
using MessagePack;

namespace Customers
{
#pragma warning disable CS8632 // The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
    [MessagePackObject]
    public class GetCustomerArgs
    {
        [Key("id")]
        public ulong Id { get; set; }
    }

    [MessagePackObject]
    public class Customer
    {
        [Key("id")]
        public ulong Id { get; set; }
        [Key("firstName")]
        public string FirstName { get; set; }
        [Key("middleName")]
        public string? MiddleName { get; set; }
        [Key("lastName")]
        public string LastName { get; set; }
        [Key("email")]
        public string Email { get; set; }
        [Key("address")]
        public Address? Address { get; set; }
    }

    [MessagePackObject]
    public class Address
    {
        [Key("line1")]
        public string Line1 { get; set; }
        [Key("line2")]
        public string? Line2 { get; set; }
        [Key("city")]
        public string City { get; set; }
        [Key("state")]
        public string State { get; set; }
        [Key("zip")]
        public string Zip { get; set; }
    }
#pragma warning restore CS8632 // The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.

    public delegate Task<Customer> CreateCustomer(Customer customer);
    public delegate Task<Customer> GetCustomer(ulong id);

    public class Inbound
    {
        public CreateCustomer CreateCustomer { get; set; }
        public GetCustomer GetCustomer { get; set; }
    }

    public interface Outbound
    {
        Task SaveCustomer(Customer customer);
        Task CustomerCreated(Customer customer);
        Task<Customer> FetchCustomer(ulong id);
    }
}

File: example/welcome/pkg/welcome/interfaces.go
package welcome

import (
	"context"
)

type ns struct{}

func (n *ns) Namespace() string {
	return "welcome.v1"
}

func (n *ns) Version() string {
	return "0.1.0"
}

type Inbound struct {
	GreetCustomer func(ctx context.Context, customer Customer) error
}

type Outbound interface {
	SendEmail(ctx context.Context, email string, message string) error
}

type Customer struct {
	ns
	FirstName string `json:"firstName" msgpack:"firstName"`
	LastName  string `json:"lastName" msgpack:"lastName"`
	Email     string `json:"email" msgpack:"email"`
}

func (c *Customer) Type() string {
	return "Customer"
}
